src/passes/OptimizeInstructions.cpp

```diff
@@ -71,7 +72,16 @@ struct PatternDatabase {
   };
 };
 
-static PatternDatabase database;
+static PatternDatabase* database = nullptr;
+
+static void ensureDatabase() {
+  if (!database) {
+    // we must only ever create one database
+    static OnlyOnce onlyOnce;
+    onlyOnce.verify();
+    database = new PatternDatabase;
```

Comment:
You don't need this only once thing since function static vars are guaranteed to be safely initialized once, on first call, even if there are threads. You can just have static PatternDatabase *database = new PatternDatabase;